### PR TITLE
Feat: Implement Amplitude Experiment infra

### PR DESCRIPTION
**Pull request by brendan-defi**
**What changed? Why?**
* Created `Experiments` context in `libs` that exposes the `experimentClient`, `getUserVariant`, and `isReady` state. 
  * Experiments initializes the Amplitude experiments infra and pulls the variants information for the current user
* Implemented the Experiments context in base-web and base-docs

**Notes to reviewers**

**How has it been tested?**
* locally, manually
* confirmed that feature flags and experimental variants are able to be retrieved from Amplitude

### CODE REVIEW

File: apps/web/src/utils/hooks/useVariant.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  flagKey: string,
+  deploymentKey = isDevelopment
+    ? defaultDeploymentKey.development
+    : defaultDeploymentKey.production,
+): string {
+  const [variant, setVariant] = useState('');
+
+  useEffect(() => {
+    try {
+      const experiment = Experiment.initialize(deploymentKey);
```
Comment (zencephalon) [inline apps/web/src/utils/hooks/useVariant.tsx:21]: Hm, I don’t think we’ll want to initialize this every time we want to use this hook. I think ideally we would reuse the one experiment client that we’ve initialized previously.

File: apps/web/src/utils/initCCA.ts [hunk 2]
```diff
@@ -44,6 +49,31 @@ const initCCA = (
     initNextJsTrackPageview({
       nextJsRouter: router,
     });
+
+    Experiment.initialize(ampDeploymentKey, {
```
Comment (zencephalon) [inline apps/web/src/utils/initCCA.ts:53]: Might as well just not do this here anymore.

Comment (brendan-defi) [reply apps/web/src/utils/initCCA.ts]: Makes sense, removed.

File: package.json [hunk 3]
```diff
@@ -76,6 +76,7 @@
   },
   "packageManager": "yarn@3.5.0",
   "dependencies": {
+    "@amplitude/experiment-js-client": "^1.11.0",
```
Comment (zencephalon) [inline package.json:79]: If anything this belongs in `base-ui`

File: apps/web/tsconfig.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
     "**/*",
     "**/*.json",
     "../../types/**/*"
-  ],
```
Comment (zencephalon) [inline apps/web/tsconfig.json:20]: This seems suspicious to me, none of the other base-ui files needed it, why would this?

File: apps/web/package.json [hunk 5]
```diff
@@ -11,6 +11,7 @@
     "update-contributors": "node ./scripts/updateContributors.js && npx prettier ./src/components/CoreContributors/CoreContributors.json -w"
   },
   "dependencies": {
+    "@amplitude/experiment-js-client": "^1.11.0",
```
Comment (zencephalon) [inline apps/web/package.json:14]: Consider moving to `base-ui`

File: package.json (hunk 3 above)
Comment (brendan-defi) [reply package.json]: roger that, removed from all `package.json` files except for `libs`

File: apps/web/tsconfig.json (hunk 4 above)
Comment (brendan-defi) [reply apps/web/tsconfig.json:20]: removed.

File: apps/web/package.json (hunk 5 above)
Comment (brendan-defi) [reply apps/web/package.json]: moved to `libs`.

**Review by zencephalon — APPROVED** [review]

**Pull request merged**